File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    # 맞힌 개수 : 점수
    rank_hash = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    min_count = 0
    
    for i in lottos:
        if i in win_nums:
            min_count += 1
    
    max_count = min_count + lottos.count(0)
    max_rank, min_rank = rank_hash[max_count], rank_hash[min_count]
    
    return [max_rank, min_rank]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.swift
import Foundation

func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var correctCount = 0
    let winNumsDict = Dictionary(uniqueKeysWithValues: zip(win_nums, Array(repeating: true, count: 6)))
                                 
    for num in lottos {
        if winNumsDict[num] != nil {
            correctCount += 1
        }
    }
    
    let zeroCount = lottos.filter { $0 == 0 }.count
    let minRank = convertToRank(for: correctCount)
    let maxRank = convertToRank(for: correctCount + zeroCount)
    
    return [maxRank, minRank]
}

func convertToRank(for correctCount: Int) -> Int {
    return correctCount >= 2 ? 7 - correctCount : 6
}